core/src/main/java/com/oath/oak/OakDetachedReadKeyBuffer.java

```diff
@@ -6,146 +6,30 @@
-    }
-
-    @Override
-    public char getChar(int index) {
-        return getTemporaryPerThreadByteBuffer().getChar(index + headerSize + position);
-    }
-
-    @Override
-    public short getShort(int index) {
-        return getTemporaryPerThreadByteBuffer().getShort(index + headerSize + position);
-    }
-
-    @Override
-    public int getInt(int index) {
-        return getTemporaryPerThreadByteBuffer().getInt(index + headerSize + position);
-    }
-
-    @Override
-    public long getLong(int index) {
-        return getTemporaryPerThreadByteBuffer().getLong(index + headerSize + position);
-    }
-
-    @Override
-    public float getFloat(int index) {
-        return getTemporaryPerThreadByteBuffer().getFloat(index + headerSize + position);
-    }
-
-    @Override
-    public double getDouble(int index) {
-        return getTemporaryPerThreadByteBuffer().getChar(index + headerSize + position);
-    }
-
-    @Override
-    public <T> T transform(OakTransformer<T> transformer) {
-        // The new ByteBuffer object is created here via slice(), to be sure that (user provided)
-        // transformer can not access anything beyond given ByteBuffer
-        ByteBuffer buffer = getTemporaryPerThreadByteBuffer().asReadOnlyBuffer();
-        if (headerSize != 0) {
-            buffer.position(buffer.position() + headerSize);
-        }
-        buffer = buffer.slice();
-        return transformer.apply(buffer);
-    }
-
-    private ByteBuffer getTemporaryPerThreadByteBuffer() {
-        // No access is allowed once the memory manager is closed.
-        // We avoid validating this here due to performance concerns.
-        // The correctness is persevered because when the memory manager is closed,
-        // its block array is no longer reachable.
-        // Thus, a null pointer exception will be raised once we try to get the byte buffer.
-        assert blockID != OakNativeMemoryAllocator.INVALID_BLOCK_ID;
-        assert position != -1;
-        assert length != -1;
-        return memoryManager.getByteBufferFromBlockID(blockID, position, length);
-    }
-
-    /*-------------- OakUnsafeDirectBuffer --------------*/
+/**
+ * This class is used for when a detached access to the key is needed without synchronization.
+ * It does not have to acquire a read lock before each access.
+ * Thus, it can only be used without other concurrent writes in the background.
+ * It is used by stream iterators that iterate over the keys (KeyStreamIterator and EntryStreamIterator).
+ */
+class OakDetachedReadKeyBuffer extends OakDetachedReadBuffer {
 
-    @Override
-    public ByteBuffer getByteBuffer() {
-        ByteBuffer buff = getTemporaryPerThreadByteBuffer().asReadOnlyBuffer();
-        buff.position(position + headerSize);
-        // The buffer's limit was set to the correct position by getTemporaryPerThreadByteBuffer()
-        return buff.slice();
-    }
+    final EntrySet.KeyBuffer key;
```

Comment:
If EntrySet.KeyBuffer is eliminated, OakAttachedReadBuffer can be used here, which actually inherits from Slice, so this way the entire "Detached-line" can avoid inheriting from Slice...